src/convolution_api.cpp

```diff
@@ -82,16 +79,9 @@ static inline auto MakeBwdCtxAndProblem(miopenHandle_t handle,
     const auto direction =
         (conv.mode != miopenTranspose) ? Direction::BackwardData : Direction::Forward;
 
-    auto problem = (conv.mode != miopenTranspose) ? ProblemDescription{miopen::deref(dyDesc),
-                                                                       miopen::deref(wDesc),
-                                                                       miopen::deref(dxDesc),
-                                                                       conv,
-                                                                       direction}
-                                                  : ProblemDescription{miopen::deref(dxDesc),
-                                                                       miopen::deref(wDesc),
-                                                                       miopen::deref(dyDesc),
-                                                                       conv,
-                                                                       direction};
+    /// \ref transpose_convolutions_x_y_swapping
+    auto problem = ProblemDescription{
+        miopen::deref(dyDesc), miopen::deref(wDesc), miopen::deref(dxDesc), conv, direction};
```

Comment:
Part 2. This is the main difference from #2476.

Other changes in this file are NFC.

Reply:
@averinevg @carlushuang FYI